### DISCUSSION TITLE
How to do testing?

### DISCUSSION THREAD

**Discussion by KuraiAndras** [stride3d/stride#1465] [Q&A] [answered]
Is there any way to run unit tests with stride? In Unity you can write tests, where you launch a scene, and then test the state of the scene after doing some actions. I have found [xunit.runner.stride](https://www.nuget.org/packages/xunit.runner.stride) on nuget, but I could not find any documentation or github issue about this topic. Is something similar to the Unity Test runner available? Maybe something something like the [AltUnity Tester?](https://altom.com/altunity/docs/altunitytester/tester.html)

**Comment by manio143** [comment, marked as answer]
You can check the sources for how test are written. For example physics test: [CharacterTest](https://github.com/stride3d/stride/blob/master/sources/engine/Stride.Physics.Tests/CharacterTest.cs) which goes over a few frames of simulation and asserts the state of entities; example visual game test: [Particles/GameTest](https://github.com/stride3d/stride/blob/master/sources/engine/Stride.Particles.Tests/GameTest.cs) it loads a predefined scene, runs 60 frames and then takes a screenshot which is later compared for a visual parity test.

**Reply by KuraiAndras** [reply]
Cool, thanks
